libraries/AP_Scripting/applets/quadplane_terrain_avoid.lua

```diff
@@ -0,0 +1,1331 @@
+
+MIN_ALT_MAX = 20
+
+REFRESH_RATE = 1.0 / (TA_UPDATE_RATE:get() or 10)
+
+local rangefinder_down_value = 0.0
+local rangefinder_forward_value = 0.0
+MAX_RANGEFINDER_VALUE = 90
+
+Q_ENABLE = Parameter('Q_ENABLE')
+Q_WVANE_ENABLE = Parameter('Q_WVANE_ENABLE')
+Q_TILT_ENABLE = Parameter('Q_TILT_ENABLE')
+THR_MAX = Parameter('THR_MAX')
+AIRSPEED_MIN = Parameter("AIRSPEED_MIN")
+AIRSPEED_CRUISE = Parameter("AIRSPEED_CRUISE")
+AIRSPEED_MAX = Parameter("AIRSPEED_MAX")
+WP_LOITER_RAD = Parameter("WP_LOITER_RAD")
+TERRAIN_ENABLE = Parameter("TERRAIN_ENABLE")
+TERRAIN_SPACING = Parameter("TERRAIN_SPACING")
+PTCH_LIM_MAX_DEG = Parameter("PTCH_LIM_MAX_DEG")
+TECS_CLMB_MAX = Parameter("TECS_CLMB_MAX")
+local airspeed_min = AIRSPEED_MIN:get() or 10
+local airspeed_cruise = AIRSPEED_CRUISE:get() or 18
+local airspeed_max = AIRSPEED_MAX:get() or 30
+local wp_loiter_rad_m = WP_LOITER_RAD:get() or 150
+local cmtc_rad_m = TA_CMTC_RAD:get() or wp_loiter_rad_m
+local terrain_spacing = TERRAIN_SPACING:get() or 100
+local ptch_lim_max_deg = PTCH_LIM_MAX_DEG:get() or 25
+local tecs_climb_max = TECS_CLMB_MAX:get() or 5.0
+
+if cmtc_rad_m <= 0 then
+    cmtc_rad_m = wp_loiter_rad_m
+end
+
+THROTTLE_CHANNEL = rc:get_channel(Parameter('RCMAP_THROTTLE'):get()) -- The RC channel used for throttle
+THROTTLE_UP_PWM = 1800
+THROTTLE_HOVER_PWM = 1500
+PITCH_CHANNEL = rc:get_channel(Parameter('RCMAP_PITCH'):get())
+PITCH_FORWARD_PWM = 1350
+PITCH_NEUTRAL_PWM = 1500
+
+PITCH_TOLERANCE = 1.1
+QUAD_TOLERANCE = 1.5
+
+local vehicle_mode = vehicle:get_mode()
+
+local current_location = ahrs:get_location()
+local current_wp_bearing_deg = 0.0
+local previous_location
+local current_altitude_m = 0.0
+local current_location_target
+local current_heading_deg = -1
+local new_location_target = Location()
+local terrain_altitude = terrain:height_above_terrain(true)
+local terrain_max_exceeded = false
+local groundspeed_vector = ahrs:groundspeed_vector()
+local groundspeed_current = groundspeed_vector:length()
+local airspeed_current = ahrs:airspeed_estimate()
+local airspeed_desired = airspeed_current
+
+local now_ms = millis()
+local old_now_ms = now_ms
+
+local pitch_last_good_timestamp_ms = now_ms
+local pitch_last_bad_timestamp_ms = now_ms
+local pitch_bad_timer = -10
+
+local slowdown_quading = false
+
+local q_wvane_enable_save = Q_WVANE_ENABLE:get()
+local avoid_enter_mode = -1
+
+-- function forward declarations
+local avoid_terrain
+local terrain_approaching
+local pitch_obstacle_detected
+local pitch_obstacle_down
+local pitch_obstacle_forward
+local quading
+
+local mavlink = require("mavlink_wrappers")
+
+local location_tracker  -- forward declaration. See below for definition and instantiation.
+
+-------------------------------------------------
+-- deal with deprecations in 4.7
+-------------------------------------------------
+version47orhigher = true
+get_yaw_function = ahrs.get_yaw_rad
+if get_yaw_function == nil then
+    ---@diagnostic disable:deprecated
```

Comment:
I think you still need the next line bit, currently this disables for the whole file I think.